### ISSUE TITLE
Merge local host id and Raft server id.

### ISSUE THREAD

**Issue by kostja** [scylladb/scylladb#9839]
Scylla master currently generates and persists two host identifiers at start: Raft Group 0 server id and UUID host id. The host id is created after joining the group 0 ring. We should merge the two, to simplify maintenance in the future. Local host id is used in system.local and system.peers as host identifier, so nodetool is already aware of it. It'd be best to just use it in Raft, but given it's created *after* raft id, it'd be best to remove it and use raft server id instead, having made it available to everybody.

**Comment by kostja** [comment]
Blocked by rewriting topology changes on Raft.

**Comment by tnozicka** [comment]
>  it'd be best to remove it and use raft server id instead, having made it available to everybody.

**Comment by tnozicka** [comment]
>  it'd be best to remove it and use raft server id instead, having made it available to everybody.

Will the RAFT server id be accessible through `system.peers` as a host_id for backward compatibility?

**Comment by bhalevy** [comment]
A pre-requisite would be to stop changing the host id of a replacing node with the replaced node host id (that caused #10148)

**Comment by bhalevy** [comment]
Is this a duplicate of https://github.com/scylladb/scylladb/issues/12204?

**Comment by kostja** [comment]
Yes.
